### PR TITLE
Adjustments for profile visibility

### PR DESCRIPTION
**Pull request by huwd**
Major changes, and big tidy up along the way to styleguide.

Basic bit is that we're doing a lot more passing of `member` and `me` down through props to do processing as close to the atomic level as possible.

More relying to the type system.

Things are much cleaner and should be easier to maintain.

### CODE REVIEW

File: src/lib/concatTitleDept.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// @flow
+import type { Member } from '../types';
+
+const concatTitleDept = (member: Member) => {
```
Comment (Floppy) [inline src/lib/concatTitleDept.js]:
```suggestion
const concatTitleOrganization = (member: Member) => {
```

File: src/lib/concatTitleDept.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  if (!jobTitle && organization) {
+    return `${organization}`;
+  }
+  if (jobTitle && !organization) {
+    return `${jobTitle}`;
+  }
+  return '';
+};
+
+export default concatTitleDept;
```
Comment (Floppy) [inline src/lib/concatTitleDept.js]:
```suggestion
export default concatTitleOrganization;
```

File: src/Component/Molecule/CardHeader/CardHeader.jsx [hunk 3]
```diff
@@ -2,11 +2,10 @@
...
 
-type CardHeaderType = {
-  /** The member's display name */
-  name: string,
-  /** The Member's Job Title */
-  jobTitle: ?string,
+import type { Member } from '../../../types';
+
+type CardHeaderProps = {
+  member: Member,
```
[the remark this answers is not in the document]
Comment (huwd) [reply src/Component/Molecule/CardHeader/CardHeader.jsx]: Resolved along with all other props

File: src/Component/Molecule/CardLocation/CardLocation.jsx [hunk 4]
```diff
@@ -4,9 +4,11 @@ import styled from 'styled-components';
 
 import MapPin from '../../Atom/MapPin/MapPin';
 
-type CardLocationTypes = {
+import type { Member } from '../../../types';
+
+type CardLocationProps = {
   /** The Member's Country */
-  country: string,
+  member: Member,
```
[the remark this answers is not in the document]
Comment (huwd) [reply src/Component/Molecule/CardLocation/CardLocation.jsx]: Resolved along with all other props

File: src/Component/Organism/ProfileBanner/ProfileBanner.jsx [hunk 5]
```diff
@@ -2,8 +2,29 @@
...
 import ProfilePicture from '../../Atom/ProfilePicture/ProfilePicture';
+import MessageButton from '../../Atom/MessageButton/MessageButton';
 
-import type { ProfileBasic } from '../../../types';
+import concatCityCountry from '../../../lib/concatCityCountry';
+import concatTitleDept from '../../../lib/concatTitleDept';
+
+import type { Member } from '../../../types';
+
+// Also used in minimal profile card
```
[the remark this answers is not in the document]
Comment (huwd) [reply src/Component/Organism/ProfileBanner/ProfileBanner.jsx]: Resolved

**Pull request merged**